### PR TITLE
Merge updatePasswor_And_recurringEvent into sprint3

### PR DESCRIPTION
**Pull request by hAnguyen1517**
Hello team.

I finished the following:

- Completed update password features.
- Merged sprint3 into my branch updatePasswor_And_recurringEvent
- Resolved merge conflicts.
- Added recurring event features.
     - Hint: Because I made changes in the models, you will need to migrate again after pulling the code in my branch.

Before working on your features, please pull my code to have the current working code in your branch to avoid any further issues!

Please let me know if you have any questions regarding my completed tasks!

Thanks!

### CODE REVIEW

File: project/home/utils.py [hunk 1]
```diff
@@ -1,61 +1,110 @@
...
-
-# home/utils.py 
-
 from datetime import datetime, timedelta
-from calendar import HTMLCalendar
+from calendar import HTMLCalendar, monthrange
 from .models import Event
-from guardian.shortcuts import get_objects_for_user
-from .templatetags.template_tags import *
-from django.urls import reverse
```
Comment (DavidmKlunk) [inline project/home/utils.py:10]: Why were these imports removed?

File: project/home/views.py [hunk 2]
```diff
@@ -76,24 +78,85 @@ def next_month(d):
...
-            messages.error(request, "You do not have permission to edit this event.")
-            return HttpResponse(status=204)
-            
-    else:
-        instance = Event()
-    
+
     form = EventForm(request.POST or None, instance=instance)
-    if request.POST and form.is_valid():
+    if request.method == 'POST' and form.is_valid():
```
Comment (DavidmKlunk) [inline project/home/views.py:86]: What was this change for?

File: project/home/views.py [hunk 3]
```diff
@@ -76,24 +78,85 @@ def next_month(d):
 
 
 def event(request, event_id=None):
-
+    instance = Event()
     if event_id:
         instance = get_object_or_404(Event, pk=event_id)
-        if not request.user.has_perm('view_event', instance):
```
Comment (DavidmKlunk) [inline project/home/views.py:82]: Why were these removed

File: project/home/models.py [hunk 4]
```diff
@@ -60,9 +71,8 @@ def get_html_url(self):
         # Override default url settings, make font black for readability
         return f'<a href="{url}" style="color: #000;">{self.title}</a>'
 
-    def save(self, *args, **kwargs):
-        super().save(*args, **kwargs)
-        # Grant 'view_event' permission to the assigned user
-        assign_perm('view_event', self.user, self)
```
Comment (DavidmKlunk) [inline project/home/models.py:66]: Why was this removed?

Comment (DavidmKlunk) [inline project/home/models.py:66]: Why was this removed?

File: project/home/utils.py (hunk 1 above)
Comment (DavidmKlunk) [inline project/home/utils.py:10]: Why were these imports removed?

File: project/home/views.py [hunk 5]
```diff
@@ -76,24 +78,85 @@ def next_month(d):
...
-            messages.error(request, "You do not have permission to edit this event.")
-            return HttpResponse(status=204)
-            
-    else:
-        instance = Event()
-
```
Comment (DavidmKlunk) [inline project/home/views.py:88]: Why were these removed

File: project/home/views.py (hunk 2 above)
Comment (DavidmKlunk) [inline project/home/views.py:86]: What was this change for?

File: project/home/models.py (hunk 4 above)
Comment (hAnguyen1517) [reply project/home/models.py:66]: My bad! Thank you for catching the issues! 
Those issues happened after I tried to resolve merge conflicts
I accidentally removed it to fix my errors. Then, I just remembered that I had to reverse back before pushing!

File: project/home/views.py (hunk 5 above)
Comment (hAnguyen1517) [reply project/home/views.py:88]: My bad! Thank you for catching the issues! 
Those issues happened after I tried to resolve merge conflicts
I accidentally removed it to fix my errors. Then, I just remembered that I had to reverse back before pushing!

File: project/home/views.py (hunk 2 above)
Comment (hAnguyen1517) [reply project/home/views.py:86]: The change works the same thing. But, because when I merged sprint3 into my branch, and I kept the current changes.

File: project/home/views.py (hunk 5 above)
Comment (hAnguyen1517) [reply project/home/views.py:88]: Because I added the recurring event, we should not pass the Event() to the instance.

File: project/home/models.py (hunk 4 above)
Comment (hAnguyen1517) [reply project/home/models.py:66]: FYI, I reversed save() back and resolved the issue!

File: project/home/utils.py (hunk 1 above)
Comment (hAnguyen1517) [reply project/home/utils.py:10]: I reversed those import back and resolved the issue!

File: project/home/views.py (hunk 5 above)
Comment (hAnguyen1517) [reply project/home/views.py:88]: Lol. I found out that because you updated sprint3 first before I'm done with my recurring event. That's when the issue occurred.
